src/openforms/submissions/logic/rules.py

```diff
@@ -154,19 +159,27 @@ def iter_evaluate_rules(
             if not triggered:
                 continue
 
-            # process all the actions in order for the triggered rule
-            for action in rule.actions:
-                action_details = action["action"]
-                # If the action type is to set a variable, update the variable state.
-                # This is the ONLY operation that is allowed to execute while we're looping
-                # through the rules.
-                if action_details["type"] == LogicActionTypes.variable:
-                    new_value = None
-                    with log_errors(action_details["value"], rule):
-                        new_value = jsonLogic(
-                            action_details["value"], data_container.data
+            for operation in map(compile_action_operation, rule.actions):
+                match operation:
+                    # only assigments to names in the context are done here
+                    case VariableAction(value=expression, variable=name):
```

Comment:
should we do any assignments at all? I wanted to decouple this as much from the action type itself (right now in master it is too coupled imo) and this introduces even tighter coupling.

It'd maybe make more sense to have the classes/instances expose an attribute `instant_evaluation: bool` to match on and take that information to do the instant evaluation. Then also make the evaluation a method of the action itself so that we keep the implementation as close to the dataclass as possible.

Reply:
So this
```py
# If the action type is to set a variable, update the variable state.
# This is the ONLY operation that is allowed to execute while we're looping
# through the rules.
if action_details["type"] == LogicActionTypes.variable:
    ...
```
should be
```py
if rule.instant_evaluation:
    ...
```
?

Reply:
yeah, I feel that makes more sense

Then in the variant for fetching remote service, we can make it an implementation detail that the result is actually saved in the database, whereas for the current "assign to variable" situation we leave them in-memory. That should be a good step towards caching that evaluation result :thinking: 

But probably best to leave that aspect for the dedicated ticket.